model/map-jpa/src/main/java/org/keycloak/models/map/storage/jpa/role/JpaRoleModelCriteriaBuilder.java

```diff
@@ -82,11 +86,29 @@ public JpaRoleModelCriteriaBuilder compare(SearchableModelField<? super RoleMode
                 }
             case IN:
                 if (modelField.equals(SearchableFields.ID)) {
-                    validateValue(value, modelField, op, String.class);
+                    if (value == null || value.length == 0) throw new CriterionNotSupportedException(modelField, op);
+
+                    final Collection collectionValues;
+                    if (value.length > 1) { // array
```

Comment:
The operand may be an array even when `value.length == 1`. For correct treatment, please see e.g. https://github.com/keycloak/keycloak/blob/4c747047cef126a182b78cb1238824b447c1fbc9/model/map/src/main/java/org/keycloak/models/map/storage/chm/CriteriaOperator.java#L144-L157

Reply:
Thank you, I haven't realized that. I've updated it according to this.